internal/stage_hp1.go

```diff
@@ -0,0 +1,101 @@
+// This test checks if the shell supports reading command history from a file using the history -r command.
+// It creates a history file with some commands, loads it into the shell, and verifies the history contents.
+
+package internal
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/codecrafters-io/shell-tester/internal/logged_shell_asserter"
+	"github.com/codecrafters-io/shell-tester/internal/shell_executable"
+	"github.com/codecrafters-io/shell-tester/internal/test_cases"
+	"github.com/codecrafters-io/shell-tester/internal/utils"
+	"github.com/codecrafters-io/tester-utils/random"
+	"github.com/codecrafters-io/tester-utils/test_case_harness"
+)
+
+func testHP1(stageHarness *test_case_harness.TestCaseHarness) error {
+	logger := stageHarness.Logger
+	shell := shell_executable.NewShellExecutable(stageHarness)
+	asserter := logged_shell_asserter.NewLoggedShellAsserter(shell)
+
+	// Step 1: Create a temporary history file with some commands
+	historyFile := filepath.Join(os.TempDir(), random.RandomWord()+"_shell_history_test")
+	defer os.Remove(historyFile)
+
+	// Set HISTFILE to /dev/null before starting the shell
+	shell.Setenv("HISTFILE", "/dev/null")
+
+	if err := asserter.StartShellAndAssertPrompt(true); err != nil {
+		return err
+	}
+
+	// Step 1: Create history file with random commands
+	nFileCommands := random.RandomInt(2, 4)
+	fileCommands := make([]string, nFileCommands)
+	for i := 0; i < nFileCommands; i++ {
+		fileCommands[i] = "echo " + strings.Join(random.RandomWords(random.RandomInt(2, 4)), " ")
+	}
+	historyFileContent := strings.Join(fileCommands, "\n") + "\n"
+	if err := os.WriteFile(historyFile, []byte(historyFileContent), 0644); err != nil {
+		logger.Errorf("Failed to write history file: %v", err)
+		return err
+	}
+
+	// print the history file content
+	content, err := os.ReadFile(historyFile)
+	if err != nil {
+		logger.Errorf("Failed to read history file: %v", err)
+		return err
+	}
+	utils.LogReadableFileContents(logger, string(content), "History file content:", filepath.Base(historyFile))
+
+	// Step 2: Run some commands in the shell
+	nShellCommands := random.RandomInt(2, 4)
+	commandTestCases := make([]test_cases.CommandResponseTestCase, nShellCommands)
+	for i := 0; i < nShellCommands; i++ {
+		cmdWords := random.RandomWords(random.RandomInt(2, 4))
+		cmd := "echo " + strings.Join(cmdWords, " ")
+		commandTestCases[i] = test_cases.CommandResponseTestCase{
+			Command:        cmd,
+			ExpectedOutput: strings.Join(cmdWords, " "),
+			SuccessMessage: fmt.Sprintf("✓ Ran %s", cmd),
+		}
+	}
+
+	// Step 3: Check history before loading from file
+	historyBefore := test_cases.HistoryTestCase{
+		CommandsBeforeHistory: commandTestCases,
+		SuccessMessage:        "✓ History before loading file is correct",
+	}
+	if err := historyBefore.Run(asserter, shell, logger); err != nil {
+		return err
+	}
+
+	// Step 4: Load history from file
+	historyLoadTest := test_cases.CommandReflectionTestCase{
```

Comment:
We'll need to be a bit more strict here. Does `CommandReflectionTestCase` actually verify that the next prompt was printed? Does it verify that no output was printed before the next prompt?

Reply:
Not 100% sure what you mean here